SwissDataScienceCenter/renga
# Notebooks service

This PR sets up the `notebooks` service (formerly an unfortunately named `projects` service) as an external jupyterhub service. 

This is a critical refactoring step, especially if we want to remove all modifications to the JH chart. In this regard, note the removal of the init container and the use of a `post-install` helm hook instead. 

A subsequent PR will remove the unrolled jupyterhub chart completely and use the helm dependency management mechanism instead. 

## Side effects

Renames `projects` --> `notebooks` in the URL used to spawn a notebook, e.g. `<jupyterhub_url>/services/notebooks/...`

## Test procedure

Lacking an integration test, you can test this on plain docker with: 

```
$ make wipe
$ PLATFORM_VERSION=development make start
```
then make a project and commit a file in gitlab. Copy the sha and go to `http://jupyterhub.renga.build/services/notebooks/<namespace>/<project>/<sha>`

On k8s minikube, follow the instructions in `charts/README.rst` (especially note the `GITLAB_URL` option passed to helm). Once everything is up, same as above (create a project + commit) then go to `$(minikube ip)/jupyterhub/services/notebooks/<namespace>/<project>/<sha>`.

Note that you should make sure that either your `.env` has `PLATFORM_VERSION=development` or set it on the command line if starting fresh. 

---

addresses #208 
addresses #146
